tests/serializers/tests.py

```diff
@@ -340,6 +341,13 @@ def test_serialize_proxy_model(self):
         self.assertEqual(base_data, proxy_data.replace('proxy', ''))
         self.assertEqual(base_data, proxy_proxy_data.replace('proxy', ''))
 
+    def test_serialize_inheritance(self):
+        child_object = Child.objects.create(parent_data=2, child_data=3)
+        child_object.parent_m2m.add(Parent.objects.create(parent_data=4))
+        child_data = serializers.serialize(self.serializer_name, [child_object])
+        self.assertNotIn('parent_data', child_data)
```

Comment:
This assertion is not related with this fix so I would move it to a separate commit.